### PR TITLE
Add duration

### PR DESCRIPTION
**Pull request by jernst**

### CODE REVIEW

**Comment by steve-bate** [comment]
Similar comment as for #158 about basing the PR branch on `develop` (for independent/orthogonal PRs). I can't merge this without merging the changes from #158 and #157.

**Comment by jernst** [comment]
I hope I didn't screw this up in rebase hell.

File: src/feditest/testruntranscript.py [hunk 1]
```diff
@@ -521,7 +521,8 @@ def _write(self):
                 nbsp=lambda s: re.sub(r'\s+', '&nbsp;', str(s)),
                 permit_line_breaks_in_identifier=lambda s: re.sub(r'(\.|::)', r'<wbr>\1', s),
                 local_name_with_tooltip=lambda n: f'<span title="{ n }">{ n.split(".")[-1] }</span>',
-                format_timestamp=lambda ts, format='%Y-%m-%dT%H-%M-%S.%fZ': ts.strftime(format) if ts else ''))
+                format_timestamp=lambda ts, format='%Y-%m-%dT%H-%M-%S.%fZ': ts.strftime(format) if ts else '',
+                format_duration=lambda d: str(d)))
```
Comment (steve-bate) [inline src/feditest/testruntranscript.py:525]: Isn't `str(d)` what jinja2 does by default if you use `{{ duration }}`?

File: src/feditest/testruntranscript.py [hunk 2]
```diff
@@ -521,7 +521,8 @@ def _write(self):
                 nbsp=lambda s: re.sub(r'\s+', '&nbsp;', str(s)),
                 permit_line_breaks_in_identifier=lambda s: re.sub(r'(\.|::)', r'<wbr>\1', s),
                 local_name_with_tooltip=lambda n: f'<span title="{ n }">{ n.split(".")[-1] }</span>',
-                format_timestamp=lambda ts, format='%Y-%m-%dT%H-%M-%S.%fZ': ts.strftime(format) if ts else ''))
+                format_timestamp=lambda ts, format='%Y-%m-%dT%H-%M-%S.%fZ': ts.strftime(format) if ts else '',
```
Comment (steve-bate) [inline src/feditest/testruntranscript.py:524]: FYI, there's also `datetime.datetime.isoformat()` that uses a format very similar or the same as this, but I understand you may have wanted the ability to optionally override the format in the template. (TIL that lambdas could have defaults for arguments. I've never seen that before.)

**Pull request merged**